presto-main/src/main/java/com/facebook/presto/operator/scalar/Centroid.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to Ted Dunning under one or more
```

Comment:
In terms of referring other open source code, I think the best practice is to keep the original license and add our license on top of the original one. Same for other files.

Reply:
An existing example in our code base is here:
[BigArrays.java](https://github.com/prestodb/presto/blob/master/presto-array/src/main/java/com/facebook/presto/array/BigArrays.java)
```java

/*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.facebook.presto.array;

// Note: this code was forked from fastutil (http://fastutil.di.unimi.it/)
// Copyright (C) 2010-2013 Sebastiano Vigna
final class BigArrays
{...}
```

Reply:
We need to ask legal people. IIRC, we cannot merge if the code specifically has this line. Actually, it might be better to depend on the jar (https://search.maven.org/search?q=a:t-digest). The package is available in most maven centrals.

Reply:
We can/should keep the original because because it's Apache v2.0 license (same as Presto). 
 We'll also need to copy over the NOTICES file that it references.

We're making some tweaks, if at the end of the day they're not needed we can pull it in from Maven central, but for now we'd like to make the changes we want in Presto.